examples/opaque-types.flix

```diff
@@ -1,12 +1,12 @@
-/// An opaque type declares a new type that is different from any other
-/// existing type. Opaque types can be used to differentiate types that
+/// An enum declares a new type that is different from any other
+/// existing type. enums can be used to differentiate types that
 /// would otherwise be the same. For example:
```

Comment:
Do we need a new explanation?

Reply:
Yeah, its fine. Maybe in the future we should consider dropping this example.